src/geo/map/cartodb-layer.js

```diff
@@ -14,7 +14,17 @@ var CartoDBLayer = LayerModelBase.extend({
     options = options || {};
 
     this._map = options.map;
-    this.bind('change:visible change:sql change:cartocss change:source', this._reloadMap, this);
+    if (attrs && attrs.cartocss) {
+      this.set('initialStyle', attrs.cartocss);
+    }
+    this.bind('change:visible change:sql change:source', this._reloadMap, this);
+    this.bind('change:cartocss', this._setCartoCSS);
```

Comment:
It looks like `_setCartoCSS` is basically doing something when cartocss has changed. I'd call it `onCartoCSSChanged` or someting along those lines...